general/device.cpp

```diff
@@ -24,21 +24,21 @@ namespace mfem
 namespace internal
 {
 
-OccaDevice occaDevice;
-
 // Backends listed by priority, high to low:
 static const Backend::Id backend_list[Backend::NUM_BACKENDS] =
 {
-   Backend::OCCA_CUDA, Backend::RAJA_CUDA, Backend::CUDA,
+   Backend::OCCA_CUDA, Backend::RAJA_CUDA, Backend::CUDA, Backend::CUDA_UVM,
    Backend::OCCA_OMP, Backend::RAJA_OMP, Backend::OMP,
-   Backend::OCCA_CPU, Backend::RAJA_CPU, Backend::CPU
+   Backend::OCCA_CPU, Backend::RAJA_CPU, Backend::DEBUG,
+   Backend::CPU
 };
 
 // Backend names listed by priority, high to low:
 static const char *backend_name[Backend::NUM_BACKENDS] =
 {
-   "occa-cuda", "raja-cuda", "cuda", "occa-omp", "raja-omp", "omp",
-   "occa-cpu", "raja-cpu", "cpu"
+   "occa-cuda", "raja-cuda", "cuda", "cuda-uvm",
+   "occa-omp", "raja-omp", "omp",
+   "occa-cpu", "raja-cpu", "debug", "cpu"
```

Comment:
Is it still true that this is the order of priority?

Reply:
the `debug` device should be exclusive: I need to add the test, you are right.
The priority should be fine then.

Reply:
Moved to the todo list